Converted PathMappings to be an AbstractMap

gregw opened the pull request:
This was primarily done to avoid surprise of the previous behavior of puts for a duplicate key being ignored. 
The use of the AbstractMap class now provides javadoc and known/expected behavior for key methods. 
The only significant behavior change is in the return type of some key methods, with the old/previous value replacing a boolean. 
Some stream usage has also been replaced by the more efficient iterator.
Some methods provided by AbstractMap may not be efficient.

gregw committed:
Converted PathMappings to be an AbstractMap

This was primarily done to avoid surprise of the previous behaviour of puts for a duplicate key being ignored.
The use of the AbstractMap class now provides javadoc and known/expected behaviour for key methods.
The only significant behaviour change is in the return type of some key methods, with the old/previous value replacing a boolean.
Some stream usage has also been replaced by the more efficient iterator.

joakime commented on jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/PathMappings.java:
```diff
@@ -41,10 +41,12 @@
-public class PathMappings<E> implements Iterable<MappedResource<E>>, Dumpable
+public class PathMappings<E> extends AbstractMap<PathSpec, E> implements Iterable<MappedResource<E>>, Dumpable
 {
+    // TODO this class should be an AbstractMap
```
Isn't this PR doing what this TODO is saying?

gregw replied:
ooops

gregw committed:
Converted PathMappings to be an AbstractMap

This was primarily done to avoid surprise of the previous behaviour of puts for a duplicate key being ignored.
The use of the AbstractMap class now provides javadoc and known/expected behaviour for key methods.
The only significant behaviour change is in the return type of some key methods, with the old/previous value replacing a boolean.
Some stream usage has also been replaced by the more efficient iterator.
